Parsers/ASimWebSession/Parsers/vimWebSessionCiscoMeraki.yaml

```diff
@@ -0,0 +1,275 @@
+Parser:
+  Title: Web Session ASIM parser for Cisco Meraki
+  Version: '0.1'
+  LastUpdated: Jun 20 2023
+Product:
+  Name: Cisco Meraki
+Normalization:
+  Schema: WebSession
+  Version: '0.2.6'
+References:
+- Title: ASIM Web Session Schema
+  Link: https://aka.ms/ASimWebSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: Cisco Meraki Documentation
+  Link: https://documentation.meraki.com/
+Description: |
+  This ASIM parser supports normalizing Cisco Meraki logs to the ASIM Web Session normalized schema.
+ParserName: vimWebSessionCiscoMeraki
+EquivalentBuiltInParser: _Im_WebSession_CiscoMeraki
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: ipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: url_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: httpuseragent_has_any
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let ActionLookup = datatable (action: string, DvcAction: string)  [
+    'allow', 'Allow',
+    'log', 'Allow',
+    'accept', 'Allow',
+    'block', 'Deny',
+    'deny', 'Deny',
+    'quarantine', 'Deny'
+  ];
+  let parser=(
+      starttime: datetime=datetime(null), 
+      endtime: datetime=datetime(null),
+      srcipaddr_has_any_prefix: dynamic=dynamic([]),
+      dstipaddr_has_any_prefix: dynamic=dynamic([]),
+      ipaddr_has_any_prefix: dynamic=dynamic([]), 
+      url_has_any: dynamic=dynamic([]),
+      httpuseragent_has_any: dynamic=dynamic([]),
+      eventresult: string='*',
+      disabled: bool=false
+      ) {
+      let src_or_any=set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let dst_or_any=set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
+      union isfuzzy=true
+          (
+          meraki_CL
+          | project-rename LogMessage =  Message
+          ),
+          (
+          Syslog
+          | project-rename LogMessage =  SyslogMessage
+          )
+      | where not(disabled) and (LogMessage has "urls" or LogMessage has_all("security_event", "security_filtering_file_scanned"))
+      | extend Parser = extract_all(@"(\d+.\d+)\s([\w\-\_]+)\s([\w\-\_]+)\s([\S\s]+)$", dynamic([1, 2, 3, 4]), LogMessage)[0]
```

Comment:
same comments applicable here as provided in ASim parser

Reply:
Event format1:
2019-07-16 04:00:30 [Local0.Info](http://local0.info/) 172.40.20.177 1 1561036264.565291108 australia_sydney security_event security_filtering_file_scanned url=http://tlu123.dl123.delivery.mp.microsoft.com/filestreamingservice/files/36ec4eb7-46dd-4aeb-990e-b6d32f7ed567?P1=1561036860& P2=402&P3=2&P4=Zj3qRDR5CbzfWlP8BuYg%2bUlTon0XE774ExEEquiawstLAJ2%2bQm3OoWLcwz3HBt8qp3r3buVRVoT5BQcUCcNlXw%3d%3d src=[172.16.10.98:64160](http://172.16.10.98:64160/) dst=[200.188.210.42:180](http://200.188.210.42:180/) mac=20:1C:BC:B2:0F:20 name='' sha256093e4fc218b27e58e2fede7b8cb044d48d66995ae785bbc186a9df5ae08ca4f7 disposition=malicious action=block

Event format2:
<134>1 1685955364.962933250 CDSAPF3_08 urls src=172.20.0.129:58066 dst=10.30.1.102:8080 mac=DC:21:48:5B:D0:1F agent='Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36' request: GET http://10.30.1.102:8080/job/Fens%20-%20Repositories/job/ui-toolkit/job/FENS-7284/wfapi/runs?since=%2313&fullStages=true&_=1685949575229

We double checked and we have the different event formats as above so we are not able to do the parsing using inbuilt function like parse to get the required fields.

Reply:
@vakohl 
As discussed in call of the last Monday because of different types of format we are not able to parse it as suggested above.

Cc: @devikamehra